game-core/src/main/java/games/strategy/triplea/delegate/BattleTracker.java

```diff
@@ -644,16 +646,15 @@ public void takeOver(final Territory territory, final PlayerID id, final IDelega
     }
     // if we have specially set this territory to have whenCapturedByGoesTo,
     // then we set that here (except we don't set it if we are liberating allied owned territory)
-    if (isTerritoryOwnerAnEnemy && newOwner.equals(id) && Matches.territoryHasWhenCapturedByGoesTo().test(territory)) {
-      for (final String value : ta.getWhenCapturedByGoesTo()) {
-        final String[] s = value.split(":");
-        final PlayerID capturingPlayer = data.getPlayerList().getPlayerId(s[0]);
-        final PlayerID goesToPlayer = data.getPlayerList().getPlayerId(s[1]);
-        if (capturingPlayer.equals(goesToPlayer)) {
+    if (isTerritoryOwnerAnEnemy
+        && newOwner.equals(id)
+        && Matches.territoryHasCaptureOwnershipChanges().test(territory)) {
+      for (final TerritoryAttachment.CaptureOwnershipChange captureOwnershipChange : ta.getCaptureOwnershipChanges()) {
+        if (captureOwnershipChange.capturingPlayer.equals(captureOwnershipChange.receivingPlayer)) {
           continue;
         }
-        if (capturingPlayer.equals(id)) {
-          newOwner = goesToPlayer;
+        if (captureOwnershipChange.capturingPlayer.equals(id)) {
+          newOwner = captureOwnershipChange.receivingPlayer;
           break;
         }
```

Comment:
Not mandatory, but this could be turned into a nice stream

Reply:
Agreed, but there is no test coverage for this code, and it's buried in a 300-line method.  I'd prefer to avoid anything more than simple replacements in this mess right now. :smile: